### PR TITLE
Allow any user to upload publications

### PR DESCRIPTION
**Pull request by AnishReddyRavula**
We would like to make it easy for users to submit publications, the backend we anyway verify the authenticity.

Tested locally and in dev (with `PermissionDenied` changes too)

image

### CODE REVIEW

File: projects/templates/projects/add_publications.html [hunk 1]
```diff
@@ -37,11 +37,23 @@ <h2>
...
 <form method="post">
-        {% csrf_token %}
-        {% bootstrap_form pubs_form %}
-        <div class="form-group">
-          <button type="submit" class="btn btn-success">Add</button>
-        </div>
-      </form>
+  {% csrf_token %}
+
+  <h4>1. Publications BibTeX Format:</h4>
```
Comment (Mark-Powers) [inline projects/templates/projects/add_publications.html:42]: I don't think these numbers are needed, we can simply just have the titles.

File: projects/templates/projects/add_publications.html [hunk 2]
```diff
@@ -37,11 +37,23 @@ <h2>
...
+  <p>Please enter your publication(s) in BibTeX format in the field below. If available, include a link to the publication using the <code>note</code> or <code>howpublished</code> field with the <a href="https://ctan.org/pkg/url">url package</a>.</p>
+  {% bootstrap_form pubs_form %}
+  <h4>2. Guidelines for Submission:</h4>
+  <ul>
+    <li>Please ensure that the publications you upload are directly related to research or projects that have utilized Chameleon resources.</li>
+    <li>Avoid duplicate submissions by checking if your publication is already listed.</li>
```
Comment (Mark-Powers) [inline projects/templates/projects/add_publications.html:48]: "listed" is unclear to me, perhaps we should link it to the papers table?

**Review by Mark-Powers — APPROVED** [review]
A few minor comments, but overall LGTM

File: projects/pub_views.py [hunk 3]
```diff
@@ -124,9 +116,7 @@ def add_publications(request, project_id):
     mapper = ProjectAllocationMapper(request)
     try:
         project = mapper.get_project(project_id)
-        if project.source != "Chameleon" or not can_add_publications(
```
[the remark this answers is not in the document]
Comment (super-cooper) [reply projects/pub_views.py:127]: It looks like this is handled by [this code](https://github.com/ChameleonCloud/portal/pull/401/files#diff-1c7c30402ba8b6607bc6f15e3bae3dbe3afdfb658b5e920c241cd841075ee553R126-R128)

[the remark this answers is not in the document]
Comment (AnishReddyRavula) [reply projects/pub_views.py:127]: Made the change to raise `PermissionDenied` if user is not part of the project

File: projects/templates/projects/add_publications.html [hunk 4]
```diff
@@ -37,11 +37,23 @@ <h2>
...
-      </form>
+  {% csrf_token %}
+
+  <h4>Publications BibTeX Format:</h4>
+  <p>Please enter your publication(s) in BibTeX format in the field below. If available, include a link to the publication using the <code>note</code> or <code>howpublished</code> field with the <a href="https://ctan.org/pkg/url">url package</a>.</p>
+  {% bootstrap_form pubs_form %}
+  <h4>Guidelines for Submission:</h4>
+  <ul>
+    <li>Please ensure that the publications you upload are directly related to research or projects that have utilized Chameleon resources.</li>
+    <li>Avoid duplicate submissions by checking if your publication is already listed <a href="https://chameleoncloud.org/user/projects/chameleon-used-research/">here.</a></li>
```
Comment (super-cooper) [inline projects/templates/projects/add_publications.html:48]: This link should also have `taget="_blank"` so we don't interrupt the form they're filling out

File: projects/templates/projects/add_publications.html [hunk 5]
```diff
@@ -37,11 +37,23 @@ <h2>
...
+    <li>In addition, it helps us a lot when you cite the Chameleon paper. Please refer <a href="https://chameleoncloud.org/learn/frequently-asked-questions/#toc-how-should-i-acknowledge-chameleon-">this FAQ</a> on how to acknowledge and cite Chameleon</li>
```
Comment (super-cooper) [inline projects/templates/projects/add_publications.html:49]: Same here `target="_blank"`

Comment (super-cooper) [inline projects/templates/projects/add_publications.html:49]: Also a small typo here

"Please refer **to** the FAQ"

File: projects/templates/projects/add_publications.html [hunk 6]
```diff
@@ -37,11 +37,23 @@ <h2>
...
+  </ul>
+
+  <p>Your cooperation in adhering to these guidelines is appreciated as it helps maintain the integrity and relevance of the publications section
```
Comment (super-cooper) [inline projects/templates/projects/add_publications.html:52]: Add a period at the end of this sentence.

**Review by super-cooper — APPROVED** [review]
Looks good mod a couple typos

**Pull request merged**